src/components/fields/derived.js

```diff
@@ -144,66 +145,63 @@ export const ContourNumeric = connectToContainer(UnconnectedNumeric, {
+    const {fullContainer} = plotProps;
     if (
-      context.container.type === 'box' &&
-      plotProps.fullValue === 'h' &&
-      context.container.y &&
-      context.container.y.length !== 0
+      plotProps.isVisible &&
+      fullContainer &&
+      fullContainer[`autobin${props.axis}`]
     ) {
-      context.updateContainer({
-        y: null,
-        ysrc: null,
-        x: context.container.y,
-        xsrc: context.container.ysrc,
-      });
+      plotProps.isVisible = false;
     }
+  },
+});
 
-    if (
-      context.container.type === 'box' &&
-      plotProps.fullValue === 'v' &&
-      context.container.x &&
-      context.container.x.length !== 0
-    ) {
-      context.updateContainer({
-        x: null,
-        xsrc: null,
-        y: context.container.x,
-        ysrc: context.container.xsrc,
-      });
-    }
+export const BinningDropdown = connectToContainer(UnconnectedDropdown, {
+  modifyPlotProps: (props, context, plotProps) => {
+    const {localize: _} = context;
+    plotProps.options =
+      plotProps.fullContainer.orientation === 'v'
+        ? [
+            {label: _('Count X'), value: 'count'},
+            {label: _('Sum Y'), value: 'sum'},
+            {label: _('Average Y'), value: 'avg'},
+            {label: _('Minimum Y'), value: 'min'},
+            {label: _('Maximum Y'), value: 'max'},
+          ]
+        : [
+            {label: _('Count Y'), value: 'count'},
+            {label: _('Sum X'), value: 'sum'},
+            {label: _('Average X'), value: 'avg'},
+            {label: _('Minimum X'), value: 'min'},
+            {label: _('Maximum X'), value: 'max'},
+          ];
+  },
+});
 
-    if (
+export const HistogramInfoVertical = connectToContainer(Info, {
+  modifyPlotProps: (props, context, plotProps) => {
+    plotProps.isVisible =
       context.container.type === 'histogram' &&
-      plotProps.fullValue === 'v' &&
-      context.container.y &&
-      context.container.y.length !== 0
-    ) {
-      context.updateContainer({
-        y: null,
-        ysrc: null,
-        ybins: null,
-        x: context.container.y,
-        xsrc: context.container.ysrc,
-        xbins: context.container.ybins,
-      });
-    }
-
-    if (
+      context.container.orientation === 'v';
+    return plotProps;
+  },
+});
+export const HistogramInfoHorizontal = connectToContainer(Info, {
+  modifyPlotProps: (props, context, plotProps) => {
+    plotProps.isVisible =
       context.container.type === 'histogram' &&
-      plotProps.fullValue === 'h' &&
-      context.container.x &&
-      context.container.x.length !== 0
-    ) {
-      context.updateContainer({
-        x: null,
-        xsrc: null,
-        xbins: null,
-        y: context.container.x,
-        ysrc: context.container.xsrc,
-        ybins: context.container.xbins,
-      });
+      context.container.orientation === 'h';
+    return plotProps;
+  },
+});
+
+export const ColorscaleNumeric = connectToContainer(UnconnectedNumeric, {
```

Comment:
what is this ?

Reply:
Oh i see, nevermind ;)

Reply:
Actually, "what is this" is a very relevant question :) 
I reread the docs and see that it's not needed. zmin and zmax do not depend on zauto - I'm removing this.